Swift 6

dimitribouniol opened the pull request:
Made Swift 6 the default, and combed through the codebase to adopt the latest language features.

I would have liked to adopt more typed throws, but for now I limited it to non-public members. We can have a separate discussion as to whether it's useful to wrap the CBOR, decoding, and cert-related errors or not (ostensibly, none of them would be the _user's_ fault, as they are receiving them from things like the browser, so I'd argue that in many cases we should wrap them? Either way, for another PR).

dimitribouniol commented on Sources/WebAuthn/WebAuthnManager.swift:
```diff
@@ -139,7 +141,7 @@ public struct WebAuthnManager: Sendable {
         timeout: Duration? = .seconds(60),
         allowCredentials: [PublicKeyCredentialDescriptor]? = nil,
         userVerification: UserVerificationRequirement = .preferred
-    ) throws -> PublicKeyCredentialRequestOptions {
```
This `throws` was completely unused. Not sure if it was here defensively or not, but removed it in case it was vestigial.

0xTim replied:
I'm guessing the challenge generator used to throw an error, but I don't think that should be an issue now

dimitribouniol commented on Sources/WebAuthn/Ceremonies/Shared/COSE/COSEAlgorithmIdentifier.swift:
```diff
@@ -19,39 +19,36 @@ import Crypto
-	// case algRS384 = -258
-	// /// AlgRS512 RSASSA-PKCS1-v1_5 with SHA-512
-	// case algRS512 = -259
-	// /// AlgPS256 RSASSA-PSS with SHA-256
-	// case algPS256 = -37
-	// /// AlgPS384 RSASSA-PSS with SHA-384
-	// case algPS384 = -38
-	// /// AlgPS512 RSASSA-PSS with SHA-512
-	// case algPS512 = -39
-	// // AlgEdDSA EdDSA
-	// case algEdDSA = -8
+    // /// AlgRS1 RSASSA-PKCS1-v1_5 with SHA-1
+    // case algRS1 = -65535
+    // /// AlgRS256 RSASSA-PKCS1-v1_5 with SHA-256
+    // case algRS256 = -257
+    // /// AlgRS384 RSASSA-PKCS1-v1_5 with SHA-384
+    // case algRS384 = -258
+    // /// AlgRS512 RSASSA-PKCS1-v1_5 with SHA-512
+    // case algRS512 = -259
+    // /// AlgPS256 RSASSA-PSS with SHA-256
+    // case algPS256 = -37
+    // /// AlgPS384 RSASSA-PSS with SHA-384
+    // case algPS384 = -38
+    // /// AlgPS512 RSASSA-PSS with SHA-512
+    // case algPS512 = -39
+    // // AlgEdDSA EdDSA
+    // case algEdDSA = -8
```
These were all tabs 🤪

0xTim replied:
Haha yeah this got picked up in #88

dimitribouniol commented on Sources/WebAuthn/Ceremonies/Shared/AuthenticatorData.swift:
```diff
@@ -43,69 +42,69 @@ extension AuthenticatorData {
-        let decoder = CBORDecoder(stream: inputStream, options: CBOROptions(maximumDepth: 16))
-        _ = try decoder.decodeItem()
+        do {
+            let decoder = CBORDecoder(stream: inputStream, options: CBOROptions(maximumDepth: 16))
+            _ = try decoder.decodeItem()
+        } catch { throw .invalidPublicKeyLength }
```
Took a chance to define a new error case for this and clarify what is happening.

0xTim commented on Sources/WebAuthn/Ceremonies/Shared/AuthenticatorData.swift:
```diff
@@ -27,11 +27,10 @@ struct AuthenticatorData: Equatable, Sendable {
-            throw WebAuthnError.authDataTooShort
-        }
+        guard bytes.count >= minAuthDataLength
+        else { throw .authDataTooShort }
```
Am not a fan of `else` being on a separate line 😆 we should get swift format integrated with an agreed style first

dimitribouniol replied:
And here I was being consistent with all my recent commits adhering to this style 🤣
Joking aside, I think this style is much more readable than the assembly of variants I replaced, as conditions can all be aligned, and the else clearly represents a one-line return or throw. When the else case has multiple statements, I agree that it looks better joined to the condition though. Either way, I've undone these fixes and will re-propose them int he context of a format PR 😄

0xTim reviewed (changes requested):
Some comments

Overall, most of the changes here are stylistic rather then related to Swift 6 so I suggest we split out the Swift 6 changes into a separate PR and have this PR as a formatting PR, integrating `swift-format` at the same time to keep everything consistent going forward (with the relevant CI checks)

dimitribouniol commented:
> so I suggest we split out the Swift 6 changes into a separate PR and have this PR as a formatting PR, integrating `swift-format` at the same time to keep everything consistent going forward (with the relevant CI checks)

I was actually prepping a separate format PR, but was waiting for in progress PRs to finish so I don't cause merge conflicts for everyone haha. I've simplified this PR to only Swift 6 features, which includes the manifest, explicit any, and typed throws where it doesn't involve much lift.
